linera-base/src/http.rs

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Types used when performing HTTP requests.
+
+use custom_debug_derive::Debug;
+use linera_witty::{WitLoad, WitStore, WitType};
+use serde::{Deserialize, Serialize};
+
+use crate::hex_debug;
+
+/// An HTTP request.
+#[derive(Clone, Debug, Eq, PartialEq, WitLoad, WitStore, WitType)]
+#[witty(name = "http-request")]
+pub struct Request {
+    /// The [`Method`] used for the HTTP request.
+    pub method: Method,
+
+    /// The URL this request is intended to.
+    pub url: String,
+
+    /// The headers that should be included in the request.
+    pub headers: Vec<Header>,
+
+    /// The body of the request.
+    #[debug(with = "hex_debug")]
+    pub body: Vec<u8>,
+}
+
+impl Request {
+    /// Creates an HTTP GET [`Request`] for a `url`.
+    pub fn get(url: impl Into<String>) -> Self {
+        Request {
+            method: Method::Get,
+            url: url.into(),
+            headers: vec![],
+            body: vec![],
+        }
+    }
+
+    /// Creates an HTTP POST [`Request`] for a `url` with a `payload` that's an arbitrary bytes.
```

Comment:
```suggestion
    /// Creates an HTTP POST [`Request`] for a `url` with a `payload` that's arbitrary bytes.
```

(or `` an arbitrary `Vec<u8>` ``?)